spec/javascript/components/lease_ups/LeaseUpApplicationsFilterContainer.test.js

```diff
@@ -30,92 +25,77 @@ const getNode = (bulkCheckboxesState = {}) => (
 )
 
-const getWrapper = (bulkCheckboxesState = {}) => mount(getNode(bulkCheckboxesState))
+const getScreen = (bulkCheckboxesState = {}) => render(getNode(bulkCheckboxesState))
 
 describe('LeaseUpApplicationsFilterContainer', () => {
   describe('status checkbox', () => {
-    let wrapper
     describe('when the list of applications is empty', () => {
       beforeEach(() => {
-        wrapper = getWrapper({})
+        getScreen({})
       })
 
       test('should render the checkbox as unchecked', () => {
-        expect(wrapper.find(Checkbox).props().checked).toBeFalsy()
-      })
-
-      test('should render the checkbox as not indeterminate', () => {
-        expect(wrapper.find(Checkbox).props().indeterminate).toBeFalsy()
+        expect(screen.getByRole('checkbox')).not.toBeChecked()
       })
 
       test('should call select all on click', () => {
         expect(mockOnClearSelectedApplications.mock.calls).toHaveLength(0)
         expect(mockOnSelectAllApplications.mock.calls).toHaveLength(0)
-        wrapper.find(Checkbox).simulate('click')
+        fireEvent.click(screen.getByRole('checkbox'))
         expect(mockOnClearSelectedApplications.mock.calls).toHaveLength(0)
         expect(mockOnSelectAllApplications.mock.calls).toHaveLength(1)
       })
     })
 
     describe('when the list of applications is all unchecked', () => {
       beforeEach(() => {
-        wrapper = getWrapper({ appId1: false, appId2: false })
+        getScreen({ appId1: false, appId2: false })
       })
 
       test('should render the checkbox as unchecked', () => {
-        expect(wrapper.find(Checkbox).props().checked).toBeFalsy()
-      })
-
-      test('should render the checkbox as not indeterminate', () => {
-        expect(wrapper.find(Checkbox).props().indeterminate).toBeFalsy()
+        expect(screen.getByRole('checkbox')).not.toBeChecked()
       })
 
       test('should call select all on click', () => {
         expect(mockOnClearSelectedApplications.mock.calls).toHaveLength(0)
         expect(mockOnSelectAllApplications.mock.calls).toHaveLength(0)
-        wrapper.find(Checkbox).simulate('click')
+        fireEvent.click(screen.getByRole('checkbox'))
         expect(mockOnClearSelectedApplications.mock.calls).toHaveLength(0)
         expect(mockOnSelectAllApplications.mock.calls).toHaveLength(1)
       })
     })
     describe('when the list of applications has some checked and some unchecked', () => {
       beforeEach(() => {
-        wrapper = getWrapper({ appId1: false, appId2: true })
-      })
-
-      test('should render the checkbox as checked=false', () => {
-        expect(wrapper.find(Checkbox).props().checked).toBeFalsy()
+        getScreen({ appId1: false, appId2: true })
       })
 
       test('should render the checkbox with indeterminate=true', () => {
-        expect(wrapper.find(Checkbox).props().indeterminate).toBeTruthy()
+        // Even though the checkbox is in the indeterminate state, it is still considered checked
```

Comment:
suggestion: make the assertion cleaner with [toBePartiallyChecked](https://github.com/testing-library/jest-dom/pull/249)

Reply:
Cool! Love that